src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Hover/HoverHandler.cs

```diff
@@ -0,0 +1,183 @@
+
+        public bool MutatesSolutionState => false;
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier? GetTextDocumentIdentifier(TextDocumentPositionParams request) => request.TextDocument;
+
+        public async Task<Hover?> HandleRequestAsync(TextDocumentPositionParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.AdditionalDocument;
+            if (document == null)
+            {
+                return null;
+            }
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var offset = text.Lines.GetPosition(ProtocolConversions.PositionToLinePosition(request.Position));
+            var textInLine = text.Lines.GetLineFromPosition(offset).ToString();
+
+            var equalPosition = textInLine.IndexOf('=');
+            var caretPosition = request.Position.Character;
+
+            // The caret may be in a position that is outside the text
+            if (caretPosition >= textInLine.Length)
+            {
+                return null;
+            }
+
+            // We don't want to show hovering description if we are over these characters
+            var character = textInLine.ElementAt(caretPosition);
+            if (character == ' ' || character == '=' || character == ',' || character == ':')
+            {
+                return null;
+            }
+
+            var description = "";
+            // We are on the left of the setting and need to display the settings name description
+            if (equalPosition != -1 && caretPosition < equalPosition)
+            {
+                var settingName = textInLine[..equalPosition].Replace(" ", "");
+                description = FindDescriptionForSetting(document, settingName);
+            }
+            // We are on the right of the setting and need to display the settings values description
+            else if (equalPosition != -1 && caretPosition > equalPosition)
+            {
+                var settingName = textInLine[..equalPosition].Replace(" ", "");
+                var colonPosition = textInLine.IndexOf(':');
+                var commaPosition = textInLine.IndexOf(',');
+
+                var settingValue = "";
+                if (colonPosition == -1 && commaPosition == -1)
+                {
+                    settingValue = textInLine[(equalPosition + 1)..].Replace(" ", "");
+                }
+                else if (colonPosition != -1)
+                {
+                    var values = textInLine[(equalPosition + 1)..].Split(':');
+                    var cont = equalPosition + 1;
+                    foreach (var element in values)
+                    {
+                        cont += element.Length + 1;
+                        if (caretPosition < cont)
+                        {
+                            settingValue = element.Replace(" ", "");
+                            break;
```

Comment:
see above comment - would allow us to just return the hover here instead of break